packages/flutter/test/material/tab_bar_theme_test.dart

```diff
@@ -54,6 +54,67 @@ void main() {
     expect(iconRenderObject.text.style.color, equals(labelColor));
   });
 
+  testWidgets('Tab bar theme overrides label styles', (WidgetTester tester) async {
+    const TextStyle labelStyle = TextStyle(fontFamily: 'foobar');
+    const TextStyle unselectedLabelStyle = TextStyle(fontFamily: 'baz');
+    const TabBarTheme tabBarTheme = TabBarTheme(
+      labelStyle: labelStyle,
+      unselectedLabelStyle: unselectedLabelStyle,
+    );
+
+    await tester.pumpWidget(_withTheme(tabBarTheme));
+
+    final RenderParagraph selectedRenderObject = tester.renderObject<RenderParagraph>(find.text(_tab1Text));
+    expect(selectedRenderObject.text.style.fontFamily, equals(labelStyle.fontFamily));
+    final RenderParagraph unselectedRenderObject = tester.renderObject<RenderParagraph>(find.text(_tab2Text));
+    expect(unselectedRenderObject.text.style.fontFamily, equals(unselectedLabelStyle.fontFamily));
+  });
+
+  testWidgets('Tab bar label styles preceed theme label styles', (WidgetTester tester) async {
+    const TextStyle labelStyle = TextStyle(fontFamily: '1');
+    const TextStyle unselectedLabelStyle = TextStyle(fontFamily: '2');
+    const TextStyle themeLabelStyle = TextStyle(fontFamily: '3');
+    const TextStyle themeUnselectedLabelStyle = TextStyle(fontFamily: '4');
+    const TabBarTheme tabBarTheme = TabBarTheme(
+      labelStyle: themeLabelStyle,
+      unselectedLabelStyle: themeUnselectedLabelStyle,
+    );
+
+    await tester.pumpWidget(
+      MaterialApp(
+          theme: ThemeData(tabBarTheme: tabBarTheme),
+          home: Scaffold(body: TabBar(
+            tabs: _tabs,
+            controller: _tabController,
+            labelStyle: labelStyle,
+            unselectedLabelStyle: unselectedLabelStyle,
+          ),
+        )
+      ),
+    );
+
+    final RenderParagraph selectedRenderObject = tester.renderObject<RenderParagraph>(find.text(_tab1Text));
+    expect(selectedRenderObject.text.style.fontFamily, equals(labelStyle.fontFamily));
+    final RenderParagraph unselectedRenderObject = tester.renderObject<RenderParagraph>(find.text(_tab2Text));
+    expect(unselectedRenderObject.text.style.fontFamily, equals(unselectedLabelStyle.fontFamily));
+  });
+
+  testWidgets('Tab bar text style defaults', (WidgetTester tester) async {
```

Comment:
Please  expand this test. It can look like the others: `_withTheme(null)` (and the _buildTabBar() function can be merged into _withTheme). Verify that that all of the colors/styles/sizes/etc have whatever values they had before this PR. The objective is to ensure that the default Tab configuration never regresses.

Note also: it would be nice if this test was first. It's the tab theme castle's outermost defenses :-)

Reply:
Done!